internal/workload/service/systemd.go

```diff
@@ -0,0 +1,168 @@
+package service
+
+import (
+	"encoding/json"
+	"io/ioutil"
+	"os"
+	"os/exec"
+	"path"
+	"sync"
+
+	"git.sr.ht/~spc/go-log"
+)
+
+const (
+	DefaultUnitsPath      = "/etc/systemd/system/"
+	DefaultRestartTimeout = 15
+	DefaultServiceSuffix  = ".service"
+	DefaultServicePrefix  = "pod-"
+)
+
+type Systemd struct {
+	Name         string            `json:"name"`
+	RestartSec   int               `json:"restartSec"`
+	Path         string            `json:"path"`
+	Units        []string          `json:"units"`
+	UnitsContent map[string]string `json:"-"`
+}
+
+type SystemdManager struct {
+	svcFilePath string
+	lock        sync.RWMutex
+	services    map[string]*Systemd
+}
+
+func NewSystemdManager(configDir string) (*SystemdManager, error) {
+	services := make(map[string]*Systemd)
+
+	servicePath := path.Join(configDir, "services.json")
+	servicesJson, err := ioutil.ReadFile(servicePath)
+	if err == nil {
+		err := json.Unmarshal(servicesJson, &services)
+		if err != nil {
+			return nil, err
+		}
+	}
+	return &SystemdManager{svcFilePath: servicePath, services: services, lock: sync.RWMutex{}}, nil
+}
+
+func (mgr *SystemdManager) Add(svc *Systemd) error {
+	mgr.lock.Lock()
+	defer mgr.lock.Unlock()
+
+	mgr.services[svc.Name] = svc
+
+	return mgr.write()
+}
+
+func (mgr *SystemdManager) Get(name string) *Systemd {
+	mgr.lock.Lock()
```

Comment:
perhaps in this case `mgr.lock.RLock()` and `mgr.lock.RULock()` will be sufficient since we should allow multiple access for reading